R/checks.R

```diff
@@ -1,3 +1,43 @@
+# Additional type checkers designed for testing argument values.
+
+check_single_logical <- function(x, call = rlang::caller_env()) {
+ cl <- match.call()
+ arg_nm <- as.character(cl$x)
+ msg <- "{.arg {arg_nm}} should be a single logical value, not {brulee:::obj_type_friendly(x)}."
+ if (!is.logical(x)) {
+  cli::cli_abort(msg, call = call)
+ }
+ if (length(x) > 1 || any(is.na(x))) {
+  cli::cli_abort(msg, call = call)
+ }
+ invisible(x)
+}
+
+check_number_whole_vec <- function(x, call = rlang::caller_env(), ...) {
+ cl <- match.call()
+ arg <- as.character(cl$x)
+
+ for (i in x) {
+  rlang:::check_number_whole(i, arg = arg, call = call, ...)
+ }
+ x <- as.integer(x)
```

Comment:
Unlike the other functions, I need an actual integer so there is coercion and a returned argument value.

Reply:
Given that this function is called `check_*()`, I wouldn't expect it to return something other than the inputted object. Maybe pull this `as.integer()` out of the function and call it where it's needed.

Reply:
Sure